tembo-io/pgmq
# feat: Add environment variable support to PGMQueue and update tests

### Description

This pull request adds environment variable support to the `PGMQueue` class and updates the tests to reflect these changes. The environment variables are loaded from a `.env` file using `python-dotenv`. This allows for easier configuration and setup of the database connection parameters.

### Changes Made

- Added `from_env` class method to `PGMQueue` to initialize settings from a `.env` file.
- Integrated `dotenv` to load environment variables.
- Updated `test_integration.py` to use the new `PGMQueue` initialization method.
- Removed unnecessary test files (`tests/conftest.py` and `tests/test_queue.py`) to clean up the test suite.
- Simplified `pyproject.toml` by removing unused dependencies and adding `python-dotenv`.
- Updated `README.md` to include instructions for setting up and using environment variables.

### How to Test

1. Ensure you have a `.env` file in the root directory with the following content:

    ```
    PG_HOST=127.0.0.1
    PG_PORT=5432
    PG_USERNAME=postgres
    PG_PASSWORD=postgres
    PG_DATABASE=test_db
    ```

2. Run the tests:

    ```sh
    python -m unittest test_integration.py    
    ```
